### ISSUE TITLE
Error with oa2df 

### ISSUE THREAD

**Issue by mamendoza1** [massimoaria/openalexR#38]
I noticed that the `oa_fetch` function sometimes errors out during the "converting" step. 

Here is my example query and the error message:

```
oa_fetch(
  entity = "works",
  title.search = "country of origin",
  publication_year = 2022,
  verbose = T
) 
```

```
Requesting url: https://api.openalex.org/works?filter=title.search%3Acountry%20of%20origin%2Cpublication_year%3A2022
Getting 1 page of results with a total of 149 records...
Error in (function (..., deparse.level = 1, make.row.names = TRUE, stringsAsFactors = FALSE,  : 
  numbers of columns of arguments do not match
```

I've done some debugging and noticed that the conversion using `oa2df` is getting stuck on 1 of the records.

**Comment by trangdata** [comment]
Thank you for reporting @mamendoza1! It was indeed an issue with oa2df, specifically works2df. I have fixed this in #39 pending approval from @massimoaria.

``` r
library(openalexR)
oa_fetch(
  entity = "works",
  title.search = "country of origin",
  publication_year = 2022
)
#> # A tibble: 153 × 28
#>    id       displ…¹ author ab    publi…² relev…³ so    so_id publi…⁴ issn  url  
#>    <chr>    <chr>   <list> <lgl> <chr>     <dbl> <chr> <chr> <chr>   <lis> <chr>
#>  1 https:/… The ex… <df>   NA    2022-0…   109.  Jour… http… Elsevi… <chr> http…
#>  2 https:/… How co… <df>   NA    2022-0…    98.8 Inte… http… Emeral… <chr> http…
#>  3 https:/… Chroni… <df>   NA    2022-0…    50.4 Hepa… http… Wiley   <chr> http…
#>  4 https:/… The in… <df>   NA    2022-0…    49.7 Jour… http… Taylor… <chr> http…
#>  5 https:/… The ef… <df>   NA    2022-0…    48.5 Unce… http… Growin… <chr> http…
#>  6 https:/… Effect… <df>   NA    2022-0…    40.8 Jour… http… Emeral… <chr> http…
#>  7 https:/… Partit… <df>   NA    2022-0…    40.4 Jour… http… Taylor… <chr> http…
#>  8 https:/… Emergi… <df>   NA    2022-0…    39.0 Jour… http… Elsevi… <chr> http…
#>  9 https:/… Nation… <df>   NA    2022-0…    37.1 Inte… http… Elsevi… <chr> http…
#> 10 https:/… Travel… <df>   NA    2022-0…    37.1 Comp… http… Taylor… <chr> http…
#> # … with 143 more rows, 17 more variables: first_page <chr>, last_page <chr>,
#> #   volume <chr>, issue <chr>, is_oa <lgl>, cited_by_count <int>,
#> #   counts_by_year <list>, publication_year <int>, cited_by_api_url <chr>,
#> #   ids <list>, doi <chr>, type <chr>, referenced_works <list>,
#> #   related_works <list>, is_paratext <lgl>, is_retracted <lgl>,
#> #   concepts <list>, and abbreviated variable names ¹​display_name,
#> #   ²​publication_date, ³​relevance_score, ⁴​publisher
```

<sup>Created on 2022-11-03 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

**Comment by mamendoza1** [comment]
Thank you @trangdata !

**Comment by mmmaat** [comment]
Hi, thanks for this package!

The bug seems to be still here, with the same input I get the exact same error as https://github.com/ropensci/openalexR/issues/38#issue-1435141506 (I just installed the package today from github).

Actually my query is a bit different but I get the same error:
```R
library(openalexR)
query = "https://api.openalex.org/works?filter=publication_year:>1989,abstract.search:populism|populist,abstract.search:economy|economic|macroeconomy|macroeconomic"
res = oa_request(query, verbose = T)
oa2df(res, entity = "works", abstract = T, verbose = T)
```

Any idea?

**Comment by trangdata** [comment]
Thank you @mmmaat for reporting this. I recently implemented a safe rbind for other entities but not _works_. I have fixed this in #109 (should be merged within the next couple days).

By the way, if you wanted to use `oa_fetch`, the result `res2` below should be the same as your `res` object. Once the PR is merged, you can re-install the package from GitHub and remove the argument `output = "list` and it should do everything you intended in 1 step.

``` r
res2 <- oa_fetch(
  "works",
  publication_year = ">1989",
  abstract.search = c("populism", "populist"),
  abstract.search = c("economy", "economic", "macroeconomy", "macroeconomic"),
  verbose = TRUE,
  abstract = TRUE,
  output = "list"
)

```
